core/test_recourse_iterator.py

```diff
@@ -0,0 +1,183 @@
+import unittest
```

Comment:
I'm skeptical about the need for this much mocking. There are alternative ways to go about setting up this testing, but given the current design complexity I think this is tolerable for now. Going forward, there is benefit to reducing complexity and simplifying wherever possible.